linker/shared/src/main/scala/org/scalajs/linker/backend/emitter/ClassEmitter.scala

```diff
@@ -45,23 +45,15 @@ private[emitter] final class ClassEmitter(sjsGen: SJSGen) {
 
   def buildClass(className: ClassName, kind: ClassKind, jsClassCaptures: Option[List[ParamDef]],
       hasClassInitializer: Boolean,
-      superClass: Option[ClassIdent], jsSuperClass: Option[Tree], useESClass: Boolean, ctor: js.Tree,
+      superClass: Option[ClassIdent], jsSuperClass: Option[Tree], useESClass: Boolean, ctorDefs: List[js.Tree],
       memberDefs: List[js.MethodDef], exportedDefs: List[js.Tree])(
       implicit moduleContext: ModuleContext,
-      globalKnowledge: GlobalKnowledge, pos: Position): WithGlobals[js.Tree] = {
+      globalKnowledge: GlobalKnowledge, pos: Position): WithGlobals[List[js.Tree]] = {
 
-    def allES6Defs = {
-      js.Block(ctor +: (memberDefs ++ exportedDefs)) match {
-        case js.Block(allDefs) => allDefs
-        case js.Skip()         => Nil
-        case oneDef            => List(oneDef)
-      }
-    }
+    def allES6Defs = ctorDefs ++ memberDefs ++ exportedDefs
```

Comment:
Any reason to use `++` instead of `:::` here? Same 3 lines below.

Reply:
No, not really (cargo culted from what was there before). Changed.